snipe/snipe-it
# Feature/custom fields default values

Implements changes suggested in #2608 (Models custom fields preloaded with data)

### This branch:
- Fixes a usage of `get()` on result of relationship (`CustomFieldsetController::fields()`), which I don't think can work? I can't find any other usage across the codebase, but have tested areas around custom fields for issues. 
- Adds ability to assign default values to custom fields for specific models
- Applies these defaults when creating an asset of this model
- Does not apply default to empty custom fields in the edit screen for models (assumption that if they are blank this is intentional and the user would not want them to fill with the default upon edit)
- Adds Bootstrap's mixin for spacing classes; overkill maybe but seemed useful for just tweaking the layout without dedicated class? Happy to remove and replace with dedicated styles

### To test:
- Create a new model and assign an existing fieldset
- Check the 'Add default fields' box
- 'Default values' box should appear below, enter some defaults
- Save model

- Create a new asset and set the model to the one created earlier
- When custom fields appear they should be pre-filled

- Edit an asset with some empty custom fields
- Custom fields that were previously empty should still be empty
- Custom fields tat were not empty should show their previous values

...only now realising it does not have ready for dev tag...oops! Will leave here anyway
